docs/60_Usage_Best_Practices.md

```diff
@@ -0,0 +1,300 @@
+
+In particular, we want to evaluate the usage of these two procedures in terms of their resilience to UPDATE/DELETE operations, which could affect the ACID properties of a collection.
+
+The analysis has produced the following results, which can be summarized in the following table.
+
+|                           | **Pagination without sorting** | **Pagination with sorting on an indexed field** | **Pagination with sorting on an immutable field (example: unique indexed field)** | **Stream with export** |
+|---------------------------|--------------------------------|------------------------------------------------|----------------------------------------------------------------------------------|------------------------|
+| PREVENTS DATA DUPLICATION |             &Cross;            |                     &Cross;                    |                                      &check;                                     |         &check;        |
+| PREVENTS DATA LOSS        |             &Cross;            |                     &Cross;                    |                                      &Cross;                                     |         &check;        |
+
+Now let's examine in detail the benefits and drawbacks of the different procedures listed.
+
+### Pagination - unordered:
+
+#### Benefits
+
+- It allows retrieving data in smaller chunks, which can be useful for displaying data in a paginated manner.
+- It provides control over the number of records fetched per page, optimizing resource utilization.
+- It can handle large result sets efficiently by retrieving only a subset of data at a time.
+
+#### Drawbacks
+
+- The order of records may not be guaranteed unless additional sorting criteria are applied.
+- If the data is frequently updated or deleted, the pagination result may become inconsistent.
+
+### Pagination - ordered:
+
+#### Benefits
+
+- It retrieves data in a specific order, providing consistency in the result set.
+- It is useful when maintaining a consistent view of data is crucial, such as in chronological or alphabetical sorting.
+
+#### Drawbacks
+
+- It may introduce performance overhead when retrieving large result sets, especially if sorting is complex.
+- If the data changes during pagination, it may lead to missing or duplicate records on subsequent pages.
+
+### Stream
+
+#### Benefits
+
+- It provides real-time data streaming, allowing applications to consume data as it becomes available.
+- It enables efficient processing and analysis of continuous data streams.
+- It can be useful in scenarios where immediate data updates are critical, such as real-time analytics or monitoring.
+
+#### Drawbacks
+
+- It requires continuous processing and handling of data streams, which may increase resource utilization.
+- Data ordering may not be guaranteed, depending on the stream source and processing pipeline.
+- In the event of disruptions or failures in stream processing, data consistency can be compromised.
```

Comment:
Does it means that during the export operation, for example not all the records are retrieved or some of them only partially?

Reply:
News on this?

Reply:
Hi @danibix95 

I verified that in the crud service, if mongodb driver fails, then an error status code is returned by the `/export` endpoint. So the last point about disruptions can be omitted.

Reply:
I added a clarification. If it is ok, I would merge the PR